UI/Pages/SearchResults.cshtml.js

```diff
@@ -0,0 +1,82 @@
+"use strict";
+
+const getAddShortlistTuitionPartnerFetchUrl = (seoUrl, localAuthority) =>
+  `?handler=AddShortlistedTuitionPartner&tuitionPartnerSeoUrl=${seoUrl}&localAuthority=${localAuthority}`;
+
+const getRemoveShortlistTuitionPartnerFetchUrl = (seoUrl) =>
+  `?handler=RemoveShortlistedTuitionPartner&tuitionPartnerSeoUrl=${seoUrl}`;
+
+const isFetchResponseDataValid = (response) =>
+  response &&
+  response.isCallSuccessful !== "undefined" &&
+  response.totalShortlistedTuitionPartners !== "undefined";
+
+const getFetchResponseData = (response) => {
+  if (response.status >= 200 && response.status <= 299) {
+    return response.json();
+  } else {
+    throw Error(response.statusText);
+  }
+};
+
+const updateSearchResultPage = (result, checkBox, elseCheckboxState) => {
+  if (isFetchResponseDataValid(result) && result.isCallSuccessful) {
+    const badgeElement = document.getElementById(
+      "totalShortlistedTuitionPartners"
+    );
+    if (badgeElement)
+      badgeElement.textContent = result.totalShortlistedTuitionPartners;
+  } else {
+    checkBox.checked = elseCheckboxState;
+  }
+};
+const onChecked = async (checkbox) => {
+  return await fetch(
+    getAddShortlistTuitionPartnerFetchUrl(
+      checkbox.id,
+      checkbox.dataset.localauthority
+    )
+  )
+    .then((response) => getFetchResponseData(response))
+    .catch((error) => {
+      //What do we want to do with the error.
+      throw Error(error);
+    });
+};
+
+const onUnChecked = async (checkbox) => {
+  return await fetch(getRemoveShortlistTuitionPartnerFetchUrl(checkbox.id))
```

Comment:
This should really be a POST rather than a GET

Reply:
Yes, I agree it should be 'POST'. This is what I was looking/saying about the Page, it does everything via 'GET' even things I think should maybe POST. I thought about refining it but the implications may be large so, I decided to stick with the current trend of using GET.